### PR TITLE
upgrade react to v18 and react pinch pan to v3

### LINKED ISSUE #3957: Upgrade to React 18

### PR DESCRIPTION
**Pull request by farooqkz**
Let me know if there is any problem.

Softly depends on removal of BluePrint as upgrading React to v18 also requires upgrading BP. And as BP will get removed, we'll have a wasted effort here.

Closes #3957

### CODE REVIEW

**Comment by farooqkz** [comment]
I don't know how to fix the remaining typing errors about `contextType`

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 - use `SOURCE_DATE_EPOCH` environment var for build timestamp instead of `Date.now()` if set.
 - use italic variants of Roboto font correctly #3949
 - show chat name when searching in chat #3950
+- upgrades react to v18 and react pinch pan zoom to v3
```
Comment (Simon-Laux) [inline CHANGELOG.md:17]: move that to the top, your change is not released yet ;)

**Comment by Simon-Laux** [comment]
This needs to be tested carefully before merging because it is a big change that might introduce new bugs.

useful links:
- https://react.dev/blog/2022/03/29/react-v18
- https://react.dev/blog/2022/03/08/react-18-upgrade-guide

typescript now needs explicit children in the props everywhere https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-typescript-definitions

**Comment by Simon-Laux** [comment]
pr looks good code wise so far :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (farooqkz) [reply CHANGELOG.md:17]: sowwy

**Comment by nicodh** [comment]
Maybe test and review again after #4006

**Comment by WofWca** [comment]
I rebased and fixed the "context" errors.

Things that don't work:
- composer's emoji picker
- Opening a profile modal prints an error
  > Warning: findDOMNode is deprecated and will be removed in the next major release.
  
I think this is because of

```log
 WARN  Issues with peer dependencies found
packages/frontend
└─┬ @blueprintjs/core 4.20.2
  └─┬ react-popper 1.3.11
    └── ✕ unmet peer react@"0.14.x || ^15.0.0 || ^16.0.0 || ^17.0.0": found 18.3.1
```

So yeah, either a Blueprint upgrade or Blueprint removal (https://github.com/deltachat/deltachat-desktop/pull/4006) is needed.

Otherwise: I checked the React 18 migration guide and I didn't notice anything that we need to do, though I didn't really understand some things from the guide.

**Comment by WofWca** [comment]
I rebased again, and fixed the emoji picker not opening issue: 26a4dc804316a60d770987d57bc3018d8158a8cb. And I suspect we might have a few more such problematic places where timing matters. I think this has to do with this item from [the React upgrade guide](https://react.dev/blog/2022/03/08/react-18-upgrade-guide):

> onsistent useEffect timing: React now always synchronously flushes effect functions if the update was triggered during a discrete user input event such as a click or a keydown event. Previously, the behavior wasn’t always predictable or consistent.

Another issue that I don't think I've seen happen outside of this MR:

https://github.com/user-attachments/assets/996ca823-63e0-4d38-a920-9ab66612fac6

I believe it has to do with this recently added code:

https://github.com/deltachat/deltachat-desktop/blob/50c5b9b89fe0946f62083dcb966529c8474accae/packages/frontend/src/components/message/MessageList.tsx#L496-L532

Other surface-level things appear to work fine.

File: packages/frontend/src/components/Gallery.tsx [hunk 2]
```diff
@@ -78,6 +78,8 @@ export default class Gallery extends Component<
     galleryImageKeepAspectRatio?: boolean
   }
 > {
+  declare context: React.ContextType<typeof DialogContext>
```
Comment (Simon-Laux) [inline packages/frontend/src/components/Gallery.tsx:81]: please add a comment where this is set

File: packages/frontend/src/components/composer/ComposerMessageInput.tsx [hunk 3]
```diff
@@ -26,6 +26,8 @@ export default class ComposerMessageInput extends React.Component<
   ComposerMessageInputProps,
   ComposerMessageInputState
 > {
+  declare context: React.ContextType<typeof DialogContext>
```
Comment (Simon-Laux) [inline packages/frontend/src/components/composer/ComposerMessageInput.tsx:29]: same question, where is this set?

**Pull request merged**